users/views.py

```diff
@@ -0,0 +1,56 @@
+from django.contrib.auth.tokens import default_token_generator
+from rest_framework import status, viewsets
+from rest_framework.decorators import action, api_view
+from rest_framework.response import Response
+from rest_framework.reverse import reverse
+
+from users.models import User
+from users.serializers import UserSerializer
+
+
+@api_view(['GET'])
+def api_root(request, format=None):
+    return Response({
+        'users': reverse('users:user_list', request=request, format=format),
+    })
+
+
+class UserViewSet(viewsets.ModelViewSet):
+    queryset = User.objects.all()
+    serializer_class = UserSerializer
+
+    def create(self, request, *args, **kwargs):
+        response = super().create(request)
+        email = request.data.get('email', '')
+        user = User.objects.get(email=email)
+        if user:
+            token = default_token_generator.make_token(user)
+            activation_link = 'activate/?user_id={}&token={}'.format(
+                    user.id, token)
+            print(activation_link)
+            url = request.build_absolute_uri(activation_link)
+            print(url)
+        else:
+            print('You need to provide a valid email address.')
```

Comment:
How does this message back to the user? How should this API user know that he needs to provide a valid email?